src/classes/CallableApi.cls

```diff
@@ -0,0 +1,144 @@
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group API
+* @group-content ../../ApexDocContent/API.htm
+* @description API class that exposes a callable class for users
+*/
+global class CallableApi implements System.Callable {
+
+    /*******************************************************************************************************
+    * @description Exception class for Malformed Method Invocations
+    ********************************************************************************************************/
+    global class MalformedMethodInvocationException extends Exception {}
+
+    /*******************************************************************************************************
+    * @description Map of valid actions used to action text
+    ********************************************************************************************************/
+    private static Map<String, String> validActionsMap = new Map<String, String> {
+        'tdtm.disablealltriggers' => 'TDTM.disableAllTriggers'
+    };
+
+    /*******************************************************************************************************
+    * @description call function implementation of the callable Interface will dispatch to the appropriate
+    * action handler based on the action text
+    * @param action string of the target action
+    * @param args name value map of arguments
+    * @return Object returned will depend on the results of the target function
+    ********************************************************************************************************/
+    global Object call(String action, Map<String, Object> args) {
+       
+        CallableApiInfo actionInfo = validateAction(action, args);
+        
+        if (actionInfo == null) {
+            throw new MalformedMethodInvocationException(action + ' is not valid');
+        }
+
+       return dispatchAction(actionInfo); 
+    }
+
+    /*******************************************************************************************************
+    * @description Helper class to keep track of the information needed for actions
+    ********************************************************************************************************/
+    public class CallableApiInfo {
```

Comment:
This is something else that I don't think adds value. A simple switch statement in the `call` methods would handle separating the actions.